build/scripts/CrossBranchPorting.psm1

```diff
@@ -0,0 +1,147 @@
+            # Create a pull request for the cherry-pick
+            gh pr create --title $title --body $body --base $TargetBranch --head $cherryPickBranch
+
+            $backportPr = gh pr view $cherryPickBranch --json url | ConvertFrom-Json
+            if ($backportPr.url) {
+                $pullRequests.Add($TargetBranch, $backportPr.url)
+            }
+        } catch {
+            RunAndCheck git checkout $startingBranch
+            Write-Host "Failed to backport to $TargetBranch. Please inspect the error and try again." -ForegroundColor Red
+            Write-Host $_.Exception.Message -ForegroundColor Red
+            throw $_.Exception.Message
+        }
+    }
+
+    # Go back to the original branch
+    RunAndCheck git checkout $startingBranch
+
+    if ($pullRequests.Count -eq 0) {
+        Write-Host "No pull requests created" -ForegroundColor Yellow
+        return
+    }
+
+    Write-Host "Backport pull requests created:" -ForegroundColor Green
+    foreach($pullRequest in $pullRequests.GetEnumerator()) {
+        Write-Host " - $($pullRequest.Key): $($pullRequest.Value)" -ForegroundColor Green
+    }
+}
+
+function PrecheckBackport($TargetBranches, $PullRequestNumber) {
+    # Check gh cli is installed
+    if (-not (Get-Command gh -ErrorAction SilentlyContinue)) {
+        throw "Please install the GitHub CLI from https://cli.github.com/"
+    }
+
+    # Check that there are no uncommitted changes
+    if (RunAndCheck git diff --name-only) {
+        throw "You have uncommitted changes. Please commit, revert or stash your changes before running this command."
+    }
+
+    # Validate Target Branches exist
+    foreach($TargetBranch in $TargetBranches) {
+        git show-ref --verify --quiet "refs/heads/$TargetBranch"
+        if ($LASTEXITCODE -ne 0) {
+            throw "Branch $TargetBranch does not exist"
+        }
+    }
+}
+
+function PortPullRequest($PullRequestDetails, $TargetBranch, $CherryPickBranch) {
+    if ((-not $PullRequestDetails.mergeCommit) -and (-not $PullRequestDetails.potentialMergeCommit)) {
+        throw "Cannot find commit to cherry-pick."
+    }
+
+    # Create a new branch for the cherry-pick
+    RunAndCheck git checkout -b $CherryPickBranch origin/$TargetBranch
+
+    # Cherry pick the merge commit
+    try {
+        if ($pullRequestDetails.mergeCommit) {
+            RunAndCheck git cherry-pick $PullRequestDetails.mergeCommit.oid
+        } else {
+            RunAndCheck git fetch origin $PullRequestDetails.potentialMergeCommit.oid
+            RunAndCheck git cherry-pick $PullRequestDetails.potentialMergeCommit.oid -m 1
+        }
+    } catch {
+        Write-Host -ForegroundColor Red "Cherry picking commitid $cherrypickId failed. $_"
+        Write-Host "To abort, press 'n' and run git cherry-pick --abort"
+        Write-Host "To continue, resolve all conflicts in a new window, run git cherry-pick --continue and then press 'y'"
+        GetConfirmation -Message "Do you want to continue?"
```

Comment:
Should `$SkipConfirmation` have a saying here? The way I get it, it's like "supress UI". Or maybe something like `ErrorAction` would be better?

Reply:
Mmh yeah but this isn't a simple "confirmation". We encountered a merge conflict and we're telling the contributor to fix it. Only once it's fixed can the porting continue. Another alternative would be to just throw an error. 
 
We can also remove the SkipConfirmation altogether. I don't mind.

Reply:
Ahh, I see. Well, whatever you think is better then.